PhillipsAuctionHouse/seldon
# fix(scss): L3-2832 scss alias map support any level of nesting

**Jira ticket**

[L3-2832](https://phillipsauctions.atlassian.net/browse/L3-2832)

**Summary**

Updates our scss mapping imports to support any level of nesting in our components to fix a bug where seldon can no longer build

**Change List (describe the changes made to the files)**

- add(buildUtils): add mapping function that can handle all the levels of nesting that our component scss files may have

**Acceptance Test (how to verify the PR)**

- Install the latest seldon in Remix
- Run `npm run build` in Remix
It will fail the build with the scss error

- Go to seldon repo
- Run `npm link`
- Run `npm run build`

- Go back to the remix project
- Run `npm link @phillips/seldon`
- Rerun `npm run build && npm run start`
- Build should succeed.

**Regression Test**

- Seldon styles should still work in Remix

**Evidence of testing**

```
> @phillipsauctionhouse/phillips-public-remix@1.6.2 build:remix
> remix vite:build

vite v5.3.3 building for production...
✓ 492 modules transformed.
build/client/.vite/manifest.json                             6.61 kB │ gzip:  0.98 kB
build/client/assets/icon-footer-red-CSro7iMd.svg            22.52 kB │ gzip:  6.87 kB
build/client/assets/Distinct-Text-xK6R82fk.woff2            29.59 kB
build/client/assets/Distinct-Italic-CM8IuJv8.woff2          29.99 kB
build/client/assets/Distinct-Display-DZP2SvOK.woff2         30.50 kB
build/client/assets/Distinct-DisplayItalic-CT4saOvC.woff2   30.56 kB
build/client/assets/image_wechat_qr_mobile-8BL5hIi1.png     35.39 kB
build/client/assets/Distinct-Text-PQWAFNZu.woff             36.18 kB
build/client/assets/Distinct-Italic-UnHqOj0i.woff           36.79 kB
build/client/assets/Distinct-Display-BC5jknC9.woff          37.11 kB
build/client/assets/Distinct-DisplayItalic-BtIXI8X3.woff    37.29 kB
build/client/assets/Montserrat-DpEGcuYQ.woff2              118.62 kB
build/client/assets/image_red_qr_mobile-DNooC11-.png       144.00 kB
build/client/assets/Montserrat-B30tuXo-.woff               156.05 kB
build/client/assets/index-CJy8AKdd.css                      58.03 kB │ gzip:  7.65 kB
build/client/assets/root-BF_6UZNV.css                       63.98 kB │ gzip:  7.71 kB
build/client/assets/careers-BkuTXYpa.js                      0.12 kB │ gzip:  0.13 kB
build/client/assets/index-Dd5boAXA.js                        0.13 kB │ gzip:  0.14 kB
build/client/assets/Button-Np8oA9a6.js                       0.38 kB │ gzip:  0.27 kB
build/client/assets/(_lang)._index-CNaSMKdp.js               1.25 kB │ gzip:  0.62 kB
build/client/assets/Link-CdBI_bV8.js                         5.61 kB │ gzip:  2.46 kB
build/client/assets/index-AxnnnQTV.js                        5.99 kB │ gzip:  2.02 kB
build/client/assets/jsx-runtime-GHZELpSI.js                  8.69 kB │ gzip:  3.21 kB
build/client/assets/i18n-CUbaX1ss.js                        10.72 kB │ gzip:  6.87 kB
build/client/assets/constants-D7zlSUQ4.js                   32.64 kB │ gzip: 11.01 kB
build/client/assets/entry.client-o1cjs5Jw.js                67.18 kB │ gzip: 20.60 kB
build/client/assets/components-Bh9us35t.js                  89.61 kB │ gzip: 29.40 kB
build/client/assets/root-D3V_iiz2.js                       106.37 kB │ gzip: 31.30 kB
build/client/assets/index-Bc_dGCJu.js                      134.36 kB │ gzip: 43.31 kB
✓ built in 1.80s
vite v5.3.3 building SSR bundle for production...
✓ 139 modules transformed.
build/server/.vite/manifest.json                             2.93 kB
build/server/assets/icon-footer-red-CSro7iMd.svg            22.52 kB
build/server/assets/Distinct-Text-xK6R82fk.woff2            29.59 kB
build/server/assets/Distinct-Italic-CM8IuJv8.woff2          29.99 kB
build/server/assets/Distinct-Display-DZP2SvOK.woff2         30.50 kB
build/server/assets/Distinct-DisplayItalic-CT4saOvC.woff2   30.56 kB
build/server/assets/image_wechat_qr_mobile-8BL5hIi1.png     35.39 kB
build/server/assets/Distinct-Text-PQWAFNZu.woff             36.18 kB
build/server/assets/Distinct-Italic-UnHqOj0i.woff           36.79 kB
build/server/assets/Distinct-Display-BC5jknC9.woff          37.11 kB
build/server/assets/Distinct-DisplayItalic-BtIXI8X3.woff    37.29 kB
build/server/assets/Montserrat-DpEGcuYQ.woff2              118.62 kB
build/server/assets/image_red_qr_mobile-DNooC11-.png       144.00 kB
build/server/assets/Montserrat-B30tuXo-.woff               156.05 kB
build/server/assets/server-build-Liv4NYYI.css               75.91 kB
build/server/index.js                                      132.80 kB
✓ built in 767ms
```



**Things to look for during review**


New Components



[L3-2832]: https://phillipsauctions.atlassian.net/browse/L3-2832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
